loki/expression/symbols.py

```diff
@@ -815,16 +819,10 @@ def __new__(cls, **kwargs):
         scope = kwargs.get('scope')
         _type = kwargs.get('type')
 
-        if scope is not None and (_type is None or _type.dtype is BasicType.DEFERRED):
+        if scope is not None and _type is None:
             # Try to determine stored type information if we have no or only deferred type
```

Comment:
The comment is out of date. Given the complexity, it might be worth expanding this here a little bit in general?

Reply:
Good spot. Should be fixed now. Thanks!